sigopt/xgboost/compute_metrics.py

```diff
@@ -1,19 +1,76 @@
 import numpy
-from sklearn.metrics import (
-  accuracy_score,
-  classification_report,
-  mean_absolute_error,
-  mean_squared_error
-)
+try:
+  from sklearn.metrics import (
+    accuracy_score,
+    classification_report,
+    mean_absolute_error,
+    mean_squared_error
+  )
+  HAS_SKLEARN = True
+except ImportError as e:
+  HAS_SKLEARN = False
```

Comment:
It is better to define all the missing metric functions here, so you do need need to change the code where metric functions are used.

Reply:
How exactly do you suggest doing this?

Reply:
something like:
`
except ImportError as e:
   def mean_absolute_error(y_true, y_pred):
     d = y_true - y_pred
     return numpy.mean(abs(d))

    
`

Reply:
I don't like this because it overloads the function and prevents easy unit testing to compare against the sklearn version.

Reply:
You can write the metric functions in another file for test, change the code about as
```
except ImportError as e:
   import utils.mean_absolute_error as mean_absolute_error
```

Reply:
I see. Let me revise everything else, write unit tests, and let's get back to this after!